executor/modes/local/handlers.py

```
@@ -369,9 +379,597 @@ async def delayed_restart():
+    def _execute_command_sync(
+        self,
+        command: str,
+        working_dir: str,
+        timeout_seconds: int,
+    ) -> Dict[str, Any]:
+        """Execute a command synchronously in a worker thread."""
+        started_at = time.monotonic()
+        resolved_working_dir = self._normalize_path(working_dir)
+
+        try:
+            result = subprocess.run(
```

Ruff (0.15.6)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)